Sources/ArcGISToolkit/Components/LocationButton/LocationButton.swift

```diff
@@ -0,0 +1,342 @@
+    var pickerText: String {
+        switch self {
+        case .off:
+            "Auto Pan Off"
+        case .recenter:
+            "Recenter"
+        case .compassNavigation:
+            "Compass"
+        case .navigation:
+            "Navigation"
+        @unknown default:
+            fatalError()
+        }
+    }
+    
+    /// The image associated with the auto pan mode.
+    var imageSystemName: String {
+        switch self {
+        case .compassNavigation:
+            "field.of.view.wide.fill"
+        case .navigation:
+            "location.north.fill"
+        case .recenter:
+            "location.fill.viewfinder"
+        case .off:
+            "location"
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+private extension Sequence where Iterator.Element: Hashable {
+    /// Returns the unique values of the sequence with the order preserved.
+    func unique() -> [Iterator.Element] {
+        var seen: Set<Iterator.Element> = []
+        return filter { seen.insert($0).inserted }
+    }
+}
+
+@MainActor
+private extension CLLocationManager {
+    static let shared = CLLocationManager()
+}
+
+#Preview {
+    @Previewable @State var locationDisplay = LocationDisplay(dataSource: SystemLocationDataSource())
+    
+    MapView(map: Map.openStreetMap())
+        .overlay(alignment: .topTrailing) {
+            VStack(spacing: 18) {
+                LocationButton(locationDisplay: locationDisplay)
+                    .padding(12)
+                    .background(.thinMaterial)
+                    .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+                    .shadow(radius: 8)
+                
+                LocationButton(locationDisplay: locationDisplay)
+                    .buttonStyle(.borderedProminent)
+                    .buttonBorderShape(.circle)
+                
+                LocationButton(locationDisplay: locationDisplay)
+                    .imageScale(.large)
+                    .frame(minWidth: 50, minHeight: 50)
+                    .background(.thinMaterial)
+                    .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+                    .shadow(radius: 8)
+                
+                LocationButton(locationDisplay: locationDisplay)
+                    .buttonStyle(.plain)
+                    .foregroundStyle(.white)
+                    .imageScale(.large)
+                    .bold()
+                    .frame(minWidth: 50, minHeight: 50)
+                    .background(.tint)
+                    .clipShape(.circle)
+                    .shadow(radius: 8)
+            }
+            .padding()
+        }
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                LocationButton(locationDisplay: locationDisplay)
+            }
+        }
+}
+
+private extension Map {
+   static func openStreetMap() -> Map {
+       Map(basemap: .init(baseLayer: OpenStreetMapLayer()))
+    }
+}
```

Comment:
This is only used the preview, but it is outside of the preview, so it will be included in release builds. Please move it inside the preview.